qabelbox/src/androidTest/java/de/qabel/qabelbox/providers/BoxProviderTest.java

```diff
@@ -183,6 +186,45 @@ public void testOpenDocumentForWrite() throws IOException, QblStorageException,
     }
 
     @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    public void testOpenDocumentForRW() throws IOException, QblStorageException, InterruptedException {
+        // Upload a test payload
+        BoxNavigation rootNav = volume.navigate();
+        byte[] testContent = new byte[] {0,1,2,3,4,5};
+        byte[] updatedContent = new byte[] {0,1,2,3,4,5,6};
+        rootNav.upload("testfile", new ByteArrayInputStream(testContent), null);
+        rootNav.commit();
+
+        // Check if the test playload can be read
+        String testDocId = ROOT_DOC_ID + "testfile";
+        Uri documentUri = DocumentsContract.buildDocumentUri(BoxProvider.AUTHORITY, testDocId);
+        assertNotNull("Could not build document URI", documentUri);
+        ParcelFileDescriptor parcelFileDescriptor =
+                mContentResolver.openFileDescriptor(documentUri, "rw");
+        FileDescriptor fileDescriptor = parcelFileDescriptor.getFileDescriptor();
+
+        InputStream inputStream = new FileInputStream(fileDescriptor);
+        byte[] dl = IOUtils.toByteArray(inputStream);
+        assertThat("Downloaded file not correct", dl, is(testContent));
+
+        // Use the same file descriptor to upload new content
+        OutputStream outputStream = new FileOutputStream(fileDescriptor);
+        assertNotNull(outputStream);
+        outputStream.write(6);
+        outputStream.close();
+        parcelFileDescriptor.close();
+
+        // wait for the upload in the background
+        // TODO: actually wait for it.
+        Thread.sleep(4000L);
```

Comment:
Well...an hour later, it still doesn't work.

The problem is that the OS calls my onClose handler which starts the upload of the file and that can take a while. Maybe I'll find a good solution tomorrow.